### PR TITLE
Grid customization to hide fields dc-87

### PR DESCRIPTION
**Pull request by snathanvj**
[](https://app.zenhub.com/workspaces/fairification-63c67b0688ee4e002b2890b3/issues/gh/ebi-ait/data-catalogue/87)

### CODE REVIEW

File: src/Catalogue.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import 'ag-grid-community/styles/ag-theme-alpine.css';
 import {fetchCatalogueData} from './api';
 import {REST_ENDPOINT_URL} from "./config";
 import {JsonSchema7 } from '@jsonforms/core';
-
+import data from './custom.json';
```
Comment (amnonkhen) [inline src/Catalogue.tsx:10]: Have you considered including this file in the config.ts file?

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: src/Catalogue.tsx [hunk 2]
```diff
@@ -16,14 +16,33 @@ interface CatalogueProps {
 const Catalogue: React.FC<CatalogueProps> = ({schema}) => {
     const [rowData, setRowData] = useState<any[]>([]);
     const [columnDefs, setColumnDefs] = useState<ColDef[]>([]);
+    const fieldConfMap = {};
+
+    //TODO: remove this as it si not required
```
Comment (amnonkhen) [inline src/Catalogue.tsx:21]: What is this comment referring to?

File: src/Catalogue.tsx [hunk 3]
```diff
@@ -16,14 +16,33 @@ interface CatalogueProps {
...
+    useEffect(() => {
+        Object.entries(data).map(([key, config]: [string, any]) => {
+            const fieldConfvalues = {};
+            Object.entries(config).map(([prop, value]: [string, any]) => {
+                // @ts-ignore
```
Comment (amnonkhen) [inline src/Catalogue.tsx:26]: What are these ts-ignore comments about?

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: src/custom.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+    "title": {
```
Comment (amnonkhen) [inline src/custom.json:2]: This file can have a few structures:

1. By Feature
```json
{
  "hide": ["col1", "col2"]
}
```

2. keys by column name - like what you did
3. Array of definitions, similarly to what ag-grid has

```json
{
   "column_definitions" : [
       {
          "name": "col1",
          "hidden": true,
          "side_filter": true
       }
   ]
}
This way you can define a typescript interface for this json object, and use it in your code to validate the input.

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: src/custom.json (hunk 4 above)
Comment (snathanvj) [reply src/custom.json:2]: Thanks, I think current structure looks simple and good. I will keep it for now.

File: src/Catalogue.tsx (hunk 2 above)
Comment (snathanvj) [reply src/Catalogue.tsx]: Removed this unused block

File: src/Catalogue.tsx (hunk 3 above)
Comment (snathanvj) [reply src/Catalogue.tsx]: it complains about json is any type.

File: src/Catalogue.tsx [hunk 5]
```diff
@@ -18,21 +19,6 @@ const Catalogue: React.FC<CatalogueProps> = ({schema}) => {
...
-            const fieldConfvalues = {};
-            Object.entries(config).map(([prop, value]: [string, any]) => {
-                // @ts-ignore
-                fieldConfvalues[prop] = value;
-            })
-            // @ts-ignore
-            fieldConfMap[key] = fieldConfvalues;
-        })
-        // @ts-ignore
-        console.log(fieldConfMap['dsIAfricaAffiliation'] + "---" + fieldConfMap['dsIAfricaAffiliation']['hide']);
```
Comment (theisuru) [inline src/Catalogue.tsx]: remove this log?

**Review by theisuru — APPROVED** [review]

File: src/Catalogue.tsx [hunk 6]
```diff
@@ -42,7 +28,7 @@ const Catalogue: React.FC<CatalogueProps> = ({schema}) => {
                     field: key,
                     editable: true,
                     // @ts-ignore
```
Comment (amnonkhen) [inline src/Catalogue.tsx:30]: Why the `@ts-ignore` directive?

File: src/Util.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import {GRID_CUSTOM_CONFIG} from './config';
+export function hide(key: string): boolean {
+    // @ts-ignore
+    return (typeof GRID_CUSTOM_CONFIG[key] === 'undefined' || GRID_CUSTOM_CONFIG[key]['hide'] === 'undefined') ? false : GRID_CUSTOM_CONFIG[key]['hide'];
```
Comment (amnonkhen) [inline src/Util.ts:4]: The 'hide' string should be a constant.

Comment (amnonkhen) [inline src/Util.ts:4]: Read about [Optional Chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html), and use here for improved readability, e.g. `GRID_CUSTOM_CONFIG?.[key]?.['hide'] ?? false`

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: src/custom.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+      "hide": false
```
Comment (amnonkhen) [inline src/custom.json:3]: Make sure `"hide": false` is optional. By default fields should be shown.

**Review by amnonkhen — CHANGES_REQUESTED** [review]

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: src/Util.ts (hunk 7 above)
Comment (snathanvj) [reply src/Util.ts:4]: Tried that already that did not work here , could not catch this undefined without typeof.

File: src/custom.json (hunk 8 above)
Comment (snathanvj) [reply src/custom.json:3]: yes , default behaviour is to show on the grid for both these cases  1)the column is not present in the custom.json 2)the "hide" key is not present for the column

File: src/Catalogue.tsx (hunk 6 above)
Comment (snathanvj) [reply src/Catalogue.tsx:30]: it complains about json is any type.

File: src/Util.ts [hunk 9]
```diff
@@ -1,5 +1,6 @@
-import {GRID_CUSTOM_CONFIG} from './config';
+import {GRID_CONFIG} from './config';
+import { HIDE_COLUMN } from './config';
 export function hide(key: string): boolean {
     // @ts-ignore
-    return (typeof GRID_CUSTOM_CONFIG[key] === 'undefined' || GRID_CUSTOM_CONFIG[key]['hide'] === 'undefined') ? false : GRID_CUSTOM_CONFIG[key]['hide'];
+    return (typeof GRID_CONFIG[key] === 'undefined' || GRID_CONFIG[key][HIDE_COLUMN] === 'undefined') ? false : GRID_CONFIG[key][HIDE_COLUMN];
```
Comment (amnonkhen) [inline src/Util.ts:5]: `HIDE_COLUMN` should be named `HIDE_COLUMN_KEY`

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: src/Util.ts (hunk 7 above)
Comment (amnonkhen) [reply src/Util.ts]: It should work. Please schedule a meeting to and we'll do it together.

**Comment by amnonkhen** [comment]
merging after a cleanup session with Senthil

**Pull request merged**